### ISSUE TITLE
ximea xiApiPython's Xi_error may not have a status value

### ISSUE THREAD

**Issue by carandraug** [python-microscope/vendor-issues#2]
Ximea's XiApiPython's returns Xi_error exception when a function returns an error code. However, if the error code is not listed on the Python side of things, Xi_error does will not have a status values. The implication of this is that ignoring certain exception types becomes more involved:

```
try:
    do_something()
except Xi_error as err:
    # Would be nice to do `if err.status == 10:`
    # Need to use getattr with default None in case status does not exist.
    if getattr(err, 'status', None) == 10:
        pass # ignore timeout (status 10)
```

The thing is that even if there's no mention of that error code on XiApiPython, there's still a status code which could be used, so this whole thing could be fixed easily.

**Comment by carandraug** [comment]
This has been reported to ximea and is ticket [55458 on ximea's ticket system](https://www.ximea.com/support/issues/55458) (which unfortunately is private).

**Comment by carandraug** [comment]
This has been fixed by Ximea since at least version "4.19.21". Not sure when it was fixed but it was still present on version "4.19.11"
